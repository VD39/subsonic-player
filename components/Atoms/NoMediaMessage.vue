<script setup lang="ts">
defineProps<{
  icon: Icon;
  message: string;
}>();
</script>

<template>
  <div :class="['column', 'centerAll', $style.noMediaMessage]">
    <component :is="icon" size="50" />

    <p>{{ message }}</p>
  </div>
</template>

<style module>
.noMediaMessage {
  flex: 1;
}
</style>
