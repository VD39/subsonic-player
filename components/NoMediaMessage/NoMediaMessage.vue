<script setup lang="ts">
defineProps<{
  icon: string;
  message: string;
}>();
</script>

<template>
  <div :class="$style.noMediaMessage">
    <div :class="$style.content">
      <component :is="icon" :size="50" />

      <p>{{ message }}</p>
    </div>
  </div>
</template>

<style module>
.noMediaMessage {
  position: relative;
  height: 100%;
}

.content {
  @mixin align-center;

  position: absolute;
  inset: 50% auto auto 50%;
  flex-direction: column;
  transform: translate(-50%, -50%);
}
</style>
