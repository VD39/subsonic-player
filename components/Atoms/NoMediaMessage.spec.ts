import type { VueWrapper } from '@vue/test-utils';

import { mount } from '@vue/test-utils';

import NoMediaMessage from './NoMediaMessage.vue';

function factory(props = {}) {
  return mount(NoMediaMessage, {
    props: {
      icon: ICONS.add,
      message: 'Message',
      ...props,
    },
  });
}

describe('NoMediaMessage', () => {
  let wrapper: VueWrapper;

  beforeEach(() => {
    wrapper = factory();
  });

  it('matches the snapshot', () => {
    expect(wrapper.html()).toMatchSnapshot();
  });
});
