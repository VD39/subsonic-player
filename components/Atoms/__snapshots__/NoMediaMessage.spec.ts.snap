// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`NoMediaMessage > matches the snapshot 1`] = `
"<div class="column centerAll noMediaMessage">
  <phpluscircle size="50"></phpluscircle>
  <p>Message</p>
</div>"
`;
